src/main/java/com/google/devtools/build/lib/skylarkbuildapi/proto/ProtoModuleApi.java

```diff
@@ -30,4 +31,12 @@
             + "to load this symbol from <a href=\"https://github.com/bazelbuild/rules_proto\">"
             + "rules_proto</a>"
             + "</p>")
-public interface ProtoModuleApi {}
+public interface ProtoModuleApi {
+  @Deprecated
+  @SkylarkCallable(
+      name = "has_protoc_do_not_use_or_we_will_break_you_without_mercy",
+      doc = "Do not use this field, its only purpose is to help with migration of "
+                + "Protobuf rules to Starlark.",
+      structField = true)
+  default void configurationFieldProtocExists() {}
```

Comment:
What's this used for? To check for the presence of other fields? Then why not check directly for their presence instead?

Reply:
This is used to check whether `--proto_compiler` is available. Unfortunately, it's the only way to feature-detect late-bound defaults. https://github.com/bazelbuild/rules_proto/blob/919671ef0f73f5a82037eaabedc39c938c13cf60/proto/private/rules/proto_toolchain.bzl#L35

Reply:
Alternative is to check for Bazel version (like we do e.g. here: https://github.com/bazelbuild/rules_rust/blob/master/rust/private/rustc.bzl#L138). That requires creating an external repository to write the bazel version.

Reply:
cc @laurentlb your thoughts on this? This is an example of users having to jump through hoops because we don't have a way to test if constant is defined?